### PR TITLE
Add dark theme to web3 modal

### PR DESCRIPTION
**Pull request by amochuko**
## Related GitHub Issue

closes #https://github.com/nation3/citizen-app/issues/310

## Screenshots (if appropriate):

_Before_ code change:
> Screenshot 2024-08-27 at 09 39 14

_After_ code change:
>Screenshot 2024-08-27 at 12 16 41

## How Has This Change Been Tested?

- [x] All [status checks](https://github.com/nation3/citizen-app/blob/main/.github/workflows/ui_mainnet.yml#L35) pass (build, lint, e2e, test)
- [x] Works on Sepolia preview deployment
- [x] Works on Mainnet preview deployment

## Are Any Admin Tasks Required?

- [x] No admin tasks

### CODE REVIEW

File: ui/components/Layout.tsx [hunk 1]
```diff
@@ -254,8 +260,12 @@ export default function Layout({ children }: any) {
 
               <p className="p-4">Connected to {connectData?.connector?.name}</p>
 
-              <ul className="menu bg-base-100 p-2 -m-2 rounded-box">
-                <li key="address">
+              <ul className="menu bg-base-100 p-2 -m-2 rounded-box space-y-2 rounded-box dark:bg-slate-800">
```
Comment (aahna-ashina) [inline ui/components/Layout.tsx:263]: If the light theme does not have vertical spacing, do not add vertical spacing to the dark theme.
```suggestion
              <ul className="menu bg-base-100 p-2 -m-2 rounded-box rounded-box dark:bg-slate-800">
```

File: ui/components/Layout.tsx [hunk 2]
```diff
@@ -254,8 +260,12 @@ export default function Layout({ children }: any) {
...
+                <li
+                  key="address"
+                  className="dark:border-[1px] dark:border-slate-600 my-1 dark:bg-slate-700
```
Comment (aahna-ashina) [inline ui/components/Layout.tsx:266]: If the light theme does not have a border, do not add a border to the dark theme.
```suggestion
                  className="my-1 dark:bg-slate-700
```

**Review by aahna-ashina — CHANGES_REQUESTED** [review]

File: ui/components/Layout.tsx [hunk 3]
```diff
@@ -297,12 +311,14 @@ export default function Layout({ children }: any) {
...
               )}
 
-              <ul className="menu bg-base-100 p-2 -m-2 rounded-box">
+              <ul className="menu bg-base-100 p-2 -m-2 rounded-box dark:bg-slate-800">
                 {connectors.map((connector) => (
                   <li key={connector.id}>
                     <button
                       disabled={!connector.ready}
                       onClick={() => connect({ connector })}
+                      className="dark:border-[1px] dark:border-slate-600 my-1 dark:bg-slate-700
```
Comment (aahna-ashina) [inline ui/components/Layout.tsx:320]:
```suggestion
                      className="dark:bg-slate-700
```

File: ui/components/Layout.tsx [hunk 4]
```diff
@@ -271,7 +281,11 @@ export default function Layout({ children }: any) {
                   </a>
                 </li>
 
-                <li key="logout">
+                <li
+                  key="logout"
+                  className="my-1 dark:bg-slate-700
+                      dark:hover:bg-slate-600 rounded-md"
```
Comment (aahna-ashina) [inline ui/components/Layout.tsx:287]:
```suggestion
                  className="dark:bg-slate-700
                      dark:hover:bg-slate-600"
```

File: ui/components/Layout.tsx [hunk 5]
```diff
@@ -254,8 +260,12 @@ export default function Layout({ children }: any) {
 
               <p className="p-4">Connected to {connectData?.connector?.name}</p>
 
-              <ul className="menu bg-base-100 p-2 -m-2 rounded-box">
-                <li key="address">
+              <ul className="menu bg-base-100 p-2 -m-2 rounded-box dark:bg-slate-800">
+                <li
+                  key="address"
+                  className="my-1 dark:bg-slate-700
+                      dark:hover:bg-slate-600 rounded-md"
```
Comment (aahna-ashina) [inline ui/components/Layout.tsx:267]:
```suggestion
                  className="dark:bg-slate-700
                      dark:hover:bg-slate-600"
```

File: ui/yarn.lock
Comment (aahna-ashina) [inline ui/yarn.lock:1]: @amochuko Please resolve the merge conflicts for this file.

**Review by aahna-ashina — CHANGES_REQUESTED** [review]
Merge conflicts in `yarn.lock`.

**Review by amochuko — COMMENTED** [review]
Kindly review the yarn.lock conflict resolve

File: ui/package.json [hunk 6]
```diff
@@ -29,7 +29,7 @@
     "eslint-config-prettier": "^9.1.0",
     "ethers": "^5.7.1",
     "jest": "^28.1.3",
-    "next": "14",
```
Comment (aahna-ashina) [inline ui/package.json:32]: @amochuko Is it necessary to downgrade Next.js?

**Comment by aahna-ashina** [comment]
> Kindly review the yarn.lock conflict resolve 

@amochuko Your branch still has merge conflicts from what I can see.

**Review by aahna-ashina — CHANGES_REQUESTED** [review]

File: ui/package.json (hunk 6 above)
Comment (amochuko) [reply ui/package.json:32]: No it's not. There isn't any need doing so.

**Review by aahna-ashina — DISMISSED** [review]

**Review by aahna-ashina — APPROVED** [review]

**Pull request merged**